File: android/http-core-java/src/main/java/com/caldremch/http/core/abs/AbsCallback.java
package com.caldremch.http.core.abs;
import org.jetbrains.annotations.Nullable;

/**
 * Created by Leon on 2022/10/30.
 */
public interface AbsCallback<T> {
    void onSuccess(@Nullable T t);
    void onError(@Nullable Throwable th);
}

File: android-http/src/main/java/com/caldremch/http/core/AbsCallback.kt
package com.caldremch.http.core

/**
 *
 * @author Caldremch
 *
 * @date 2020-01-09 15:14
 *
 * @email caldremch@163.com
 *
 * @describe
 *
 **/
interface AbsCallback<T> {
    /**
     * 成功回调返回
     * @param data 具体解析类型
     */
    fun onSuccess(data: T?)

    fun onError(e: Throwable?) {}
}